generate_isoscape.py

```diff
@@ -0,0 +1,191 @@
+  if metadata.get(gdal.DCAP_CREATECOPY) != "YES":
+      raise RuntimeError("GTiff driver does not support required method CreateCopy().")
+
+  dataset = driver.Create(path, bounds.raster_size_x, bounds.raster_size_y, prediction.shape[2], eType=gdal.GDT_Float64)
+  dataset.SetGeoTransform([bounds.minx, bounds.pixel_size_x, 0, bounds.maxy, 0, bounds.pixel_size_y])
+  dataset.SetProjection('GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0],UNIT["degree",0.0174532925199433],AUTHORITY["EPSG","4326"]]')
+
+  if len(prediction.shape) != 3 or prediction.shape[0] != bounds.raster_size_x or prediction.shape[1] != bounds.raster_size_y:
+    raise ValueError("Shape of prediction does not match base geotiff")
+  
+  prediction_transformed = np.flip(np.transpose(prediction, axes=[1,0,2]), axis=0)
+  for band_index in range(dataset.RasterCount):
+    band = dataset.GetRasterBand(band_index+1)
+    if band.CreateMaskBand(0) == gdal.CE_Failure:
+      raise RuntimeError("Failed to create mask band")
+    mask_band = band.GetMaskBand()
+    band.WriteArray(np.choose(prediction_transformed[:, :, band_index].mask, (prediction_transformed[:, :, band_index].data,np.array(band.GetNoDataValue()),)))
+    mask_band.WriteArray(np.logical_not(prediction_transformed[:, :, band_index].mask))
+
+def get_predictions_at_each_pixel(
+    model: model.Model,
+    geotiffs: dict[str, raster.AmazonGeoTiff],
+    bounds: raster.Bounds,
+    geometry_mask: raster.AmazonGeoTiff=None):
+  """Uses `model` to make mean/variance predictions for every pixel in `bounds`.
+  Queries are constructed by querying every geotiff in `geotiffs` for information 
+  at that pixel and passing the parameters to the model. 
+  
+  Parameters are standardized using feature_transformer, a set of standardizers used
+  to fit training data.
+  
+  `model`: Tensorflow model used to make predictions
+  `feature_transformer`: ScikitLearn ColumnTransformer storing transformations of columns
+                         Input must be transformed prior to predictions
+  `geotiffs`: The set of geotiffs required to make the prediction
+  `bounds`: Every pixel within these bounds will have a prediction made on it
+  `geometry_mask`: If specified, only make predictions within this mask and within `bounds`."""
+
+  # Initialize a blank plane representing means and variance.
+  predicted_np = np.ma.array(
+      np.zeros([bounds.raster_size_x, bounds.raster_size_y, 2], dtype=float),
+      mask=np.ones([bounds.raster_size_x, bounds.raster_size_y, 2], dtype=bool))
+
+  for x_idx, x in enumerate(tqdm(np.arange(bounds.minx, bounds.maxx, bounds.pixel_size_x, dtype=float))):
+    rows = []
+    row_indexes = []
+    for y_idx, y in enumerate(np.arange(bounds.miny, bounds.maxy, -bounds.pixel_size_y, dtype=float)):
+      # Row should contain all the features needed to predict, in the same
```

Comment:
I'm having a hard time figuring out how to meaningfully validate this. The column order is enforced by the order of the `geotiffs` dictionary. I feel like checking if it matches the order in the model's ColumnTransformer is redundant as `geotiffs` is created by iterating over all the columns in ColumnTransformer + the dictionary's order is preserved.

Reply:
Since this is an expensive function, it seems like it should be possible to check that the order of `geotiffs.keys()` matches the order of the feature keys listed by the model. This stuff tends to break over time, and having a check at the top of the function would preempt that (or even better-- extracting the check into a function and using it to guard API calls).